app/assets/javascripts/request_viewer_state.js

```diff
@@ -0,0 +1,93 @@
+                    const canvas_id = windows[viewer]['canvasId'];
+                    const iiif_initial_viewer_config = JSON.stringify(viewers[viewer]);
+                    const canvasField = document.querySelector(`#${this.formId} > input[name="item[${this.itemId}][iiif_canvas_id]"]`)
+                    const configInput = document.querySelector(`#${this.formId} > input[name="item[${this.itemId}][iiif_initial_viewer_config]"]`)
+                    canvasField.value = canvas_id;
+                    configInput.value = iiif_initial_viewer_config;
+                    document.querySelector(`#${this.formId} > input[name="item[${this.itemId}][full_image_url]"]`).value = iiif_images[0];
+                    const thumbnail_size = iiif_images.length > 1 ? '!33,100' : '!100,100';
+                    const thumbnail = iiif_images.map(image => image.replace('full', thumbnail_size))
+                    document.querySelector(`#${this.formId} > input[name="item[${this.itemId}][thumbnail_image_url]"]`).value = thumbnail[0];
+                    document.querySelector(`#${this.formId} img`).src = `${thumbnail[0]}?${new Date().getTime()}`;
+                    if (thumbnail.length > 1){
+                      if (!document.querySelector('#page-2')) {
+                      document.querySelector(`#${this.formId} img`).insertAdjacentHTML('afterend', `<img id="page-2" src="${thumbnail[1]}?${new Date().getTime()}"/>`);
+                      } else {
+                        document.querySelector('#page-2').src = `${thumbnail[1]}?${new Date().getTime()}`;
+                      }
+                    } else if (document.querySelector('#page-2')) {
+                      document.querySelector('#page-2').src = ''
+                    }
+                    let modal_link_element = document.querySelector(`#${this.formId} #select-image-area`);
+                    let modal_link = this.updateQueryParameters(modal_link_element.href, canvas_id, iiif_initial_viewer_config);
+                    const multi_page = document.querySelector('[data-current-image]');
+                    if (multi_page) multi_page.innerText = canvas_index;
+                    modal_link_element.href = modal_link;
+                }
+              }
+          });
+      },
+
+      updateQueryParameters: function(url, canvas_id, iiif_initial_viewer_config) {
+        const urlObj = new URL(url);
+        urlObj.searchParams.set('canvas_id', canvas_id);
+        urlObj.searchParams.set('iiif_initial_viewer_config', iiif_initial_viewer_config);
+        return urlObj.href;
+      },
+
+      requestState: function() {
+          const iframe = document.querySelector('.oembed-widget iframe, iframe.mirador-embed-wrapper');
+          iframe.contentWindow.postMessage(JSON.stringify({ type: 'requestState' }), '*'); // Change '*' to a specific origin for security?
+      }
+  };
+
+  global.RequestViewerState = RequestViewerState;
```

Comment:
Why do we need to set this? Can we just import the `init` function?

Reply:
This seems to be how we are doing it in all our JS. If you have specific code I can plug in to test to see if it works then I will do that, but otherwise I do not think it is worth the time it will take me to figure it out. Especially since this is inline with what we are doing in this application. 

https://github.com/sul-dlss/exhibits/blob/main/app/assets/javascripts/bibliography.js#L101
https://github.com/sul-dlss/exhibits/blob/main/app/assets/javascripts/nested_related_items.js#L86

Reply:
All of this JS was written 10 years ago, pre ES2015 standard.  Just because it's been that way in the past, doesn't mean we should repeat those patterns.

Reply:
If you have specific code I can plug in to test to see if it works then I will do that, but otherwise I do not think it is worth the time it will take me to figure it out. We only have til the end of this week to get this into exhibits and it is a priority.